src/Agent.Worker/TaskDefinitionManager.cs

```diff
@@ -0,0 +1,291 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
```

Comment:
This file was named TaskManager. I felt like the name was confusing since we also have TaskRunner.

This class is for managing the task definitions: 1) task json containing default input values and handler metadata 2) actual task scripts.

TaskPackageManager might be an even better name. I would need to rename the Load method to LoadDefinition. The DownloadAsync method name would still make sense.